src/Hosting/Abstractions/src/BackgroundService.cs

```diff
@@ -29,11 +29,49 @@ public abstract class BackgroundService : IHostedService, IDisposable
         /// <param name="cancellationToken">Indicates that the start process has been aborted.</param>
         public virtual Task StartAsync(CancellationToken cancellationToken)
         {
-            // Create linked token to allow cancelling executing task from provided token
-            _stoppingCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            CancellationToken stoppingCtsToken;
 
+            // Only if _stoppingCts not created before or created and canceled
+            if (_stoppingCts?.IsCancellationRequested != false)
+            {
+                _stoppingCts = CancellationTokenSource.CreateLinkedTokenSource();
```

Comment:
shame, shame on me! this line need to fix `_stoppingCts = new CancellationTokenSource();`